File: week1/day5_first_unique_character_in_a_string/first.cpp
class Solution {
  public:
    int firstUniqChar(string s) {
        unordered_map<char, int> counter;
        int c = 0;
        for (char &ch : s) {
            counter[ch]++;
            c++;
        }
        c = 0;
        for (char &ch : s) {
            if (counter[ch] == 1) {
                return c;
            }
            c++;
        }
        return -1;
    }
};

File: week4/day1_sort_characters_by_frequency/first.py
from collections import Counter
from functools import cmp_to_key


class Solution:
    def frequencySort(self, s: str) -> str:
        c = Counter(s)

        def func(x, y):
            if c[y] > c[x]:
                return 1
            elif c[y] == c[x]:
                return 1 if y > x else -1
            else:
                return -1

        return "".join(sorted(s, key=cmp_to_key(func)))


